viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -595,7 +598,36 @@ void CartoFacade::GetPointCloudMap(viam_carto_get_point_cloud_map_response *r) {
     r->point_cloud_pcd = to_bstring(pointcloud_map);
 };
 
-void CartoFacade::GetInternalState(viam_carto_get_internal_state_response *r){};
+// TODO: This function is unnecessarily prone to IO errors
+// due to going through the file system  in order to read
+// the internal state.
+// This is the ticket to remove that failure mode:
+// https://viam.atlassian.net/browse/RSDK-3878
```

Comment:
I am seeing that it indeed does look silly to write the internal state to a temporary file, read it back out, and delete the temporary file :) Feel free to resolve.